verification/src/compiler/version.rs

```diff
@@ -209,4 +253,93 @@ mod tests {
             )
         );
     }
+
+    #[test]
+    fn order_versions() {
+        let versions = vec![
+            "solc-v0.5.2-nightly.2018.12.7+commit.52ff3c94",
+            "solc-v0.5.2-nightly.2018.12.6+commit.5a08ae5e",
+            "solc-v0.5.2-nightly.2018.12.5+commit.6efe2a52",
+            "solc-v0.5.2-nightly.2018.12.4+commit.e49f37be",
+            "solc-v0.5.2-nightly.2018.12.3+commit.e6a01d26",
+            "solc-v0.5.2-nightly.2018.12.19+commit.88750920",
+            "solc-v0.5.2-nightly.2018.12.18+commit.4b43aeca",
+            "solc-v0.5.2-nightly.2018.12.17+commit.12874029",
+            "solc-v0.5.2-nightly.2018.12.13+commit.b3e2ba15",
+            "solc-v0.5.2-nightly.2018.12.12+commit.85291bcb",
+            "solc-v0.5.2-nightly.2018.12.11+commit.599760b6",
+            "solc-v0.5.2-nightly.2018.12.10+commit.6240d9e7",
+            "solc-v0.5.2+commit.1df8f40c",
+            "solc-v0.5.17+commit.d19bba13",
+            "solc-v0.5.16+commit.9c3226ce",
+            "solc-v0.5.15+commit.6a57276f",
+            "solc-v0.5.14-nightly.2019.12.9+commit.d6667560",
+            "solc-v0.6.3-nightly.2020.1.31+commit.b6190e06",
+            "solc-v0.6.3-nightly.2020.1.30+commit.ad98bf0f",
+            "solc-v0.6.3-nightly.2020.1.29+commit.01eb9a5b",
+            "solc-v0.6.3-nightly.2020.1.28+commit.2d3bd91d",
+            "solc-v0.6.3-nightly.2020.1.27+commit.8809d4bb",
+            "solc-v0.6.3+commit.8dda9521",
+            "solc-v0.6.2+commit.bacdbe57",
+            "solc-v0.6.12+commit.27d51765",
+            "solc-v0.6.2-nightly.2020.1.9+commit.17158995",
+            "solc-v0.6.2-nightly.2020.1.8+commit.12b52ae6",
+            "solc-v0.6.2-nightly.2020.1.27+commit.1bdb409b",
+            "solc-v0.6.2-nightly.2020.1.23+commit.3add37a2",
+            "solc-v0.6.2-nightly.2020.1.22+commit.641bb815",
+            "solc-v0.6.2-nightly.2020.1.20+commit.470c19eb",
+            "solc-v0.6.2-nightly.2020.1.17+commit.92908f52",
+            "solc-v0.6.2-nightly.2020.1.16+commit.3d4a2219",
+            "solc-v0.6.2-nightly.2020.1.15+commit.9d9a7ebe",
+            "solc-v0.6.2-nightly.2020.1.14+commit.6dbadf69",
+            "solc-v0.6.2-nightly.2020.1.13+commit.408458b7",
+        ];
+        let mut versions: Vec<CompilerVersion> = versions
+            .iter()
+            .map(|s| CompilerVersion::from_str(s).expect("invalid version"))
+            .collect();
+        versions.sort();
+        let versions: Vec<String> = versions.iter().map(|v| v.to_string()).collect();
+        assert_eq!(
```

Comment:
give an already sorted array and check `assert_eq!(versions, sorted)`
then also random shuffle it like 10-100 times and check `assert_eq!(versions, shuffled_then_sorted)`

Reply:
but actually it would be better to have 3 independent tests, which compares
`release <=> release`
`nightly <=> release`
`release <=> nightly`
and in each test to have several comparisons (at least one for each Ordering)

Reply:
I support with the idea with 3 tests one per each possible case. But shouldn't the last one be `nightly <=> nightly` instead of `release <=> nightly`?